src/main/java/Cart.java

```diff
@@ -0,0 +1,80 @@
+import exceptions.NotFoundInArrayList;
+
+import java.util.ArrayList;
+
+public class Cart {
+    private ArrayList<CartItem> items = new ArrayList<>();
+    private int deliveryFee;
+    
+   public void setDeliveryFee(int deliveryFee) {
+        this.deliveryFee = deliveryFee;
+    }
+
+    public int getDeliveryFee() {
+        return deliveryFee;
+    }
+    
+    public Cart(){
+    }
+
+    //Calculate the VAT
+    private Double calculateVAT(double itemFullPrice){
+        return itemFullPrice * 0.2 ;
+    }
+
+    //Calculate the final price of all products
+    private Double calculateFinalPrice(){
+        double sum = 0;
+        for (var item: items) {
+            sum += item.calculateTheFinalSumWithQuantity(item.getQuantity());
+        }
+        if(sum > 0) {
+            if (sum < 100) {
+                setDeliveryFee(10);
+            } else if (sum < 200) {
+                setDeliveryFee(5);
+            } else {
+                setDeliveryFee(0);
+            }
+        }else{
+            System.out.println("Your cart is empty. Please add an item to calculate!");
+        }
+        return (sum + getDeliveryFee()) - calculateVAT(sum);
```

Comment:
Final price is without VAT?

Reply:
My fault, I already added VAT to the final sum.